### ISSUE TITLE
How to install on linux

### ISSUE THREAD

**Issue by gregoryperrin** [Denvi/grblControl#25]
I assumed I had to download and extract the tar.gz linux version, and then execute the grblcontrol file. When i do it the program opens, i select the port, it says "connected" but i can't get the CNC to move (shapeoko 3). Is there some dependencies/librairies i am missing?

I'm on linux mint rafaela 64bits. I've been succesfully been able to use my CNC with other grbl controllers in the last two weeks.

Any clue?

Thank you,

Gregory

**Comment by Denvi** [comment]
"Connected" means port is opened so all dependencies in place. 

In normal case you should see "Ready" state after program start.
Looks like it's problem with port name/baud rate.

What baud rate you use? Try to increase "Query period" in settings.

**Comment by gregoryperrin** [comment]
I used 115,200. I will to increase "Query period" in settings.

**Comment by gregoryperrin** [comment]
I increased, no success. I entered 50, 60, 100, 3000.

**Comment by Denvi** [comment]
Post please screenshots:
1) Main window of "grblControl" program after starting.
2) Settings window with "Connection" group visible.
3) Same for other application, which works with your CNC.

**Comment by gregoryperrin** [comment]
Hi Denvi, here are the screenshots

main window
----------------
![main_window](https://cloud.githubusercontent.com/assets/15223909/11831914/3a6dc7fc-a381-11e5-8082-65594e739e26.png)

settings window
--------------------
![settings_window](https://cloud.githubusercontent.com/assets/15223909/11831915/40430296-a381-11e5-8b53-40d7954ecfe2.png)
Some Universal g-code Sender screenshots
--------------------------------------------------------
![ugs](https://cloud.githubusercontent.com/assets/15223909/11831916/4050810a-a381-11e5-9d9b-204838359f66.png)
![ugs02](https://cloud.githubusercontent.com/assets/15223909/11831917/405756a6-a381-11e5-8a56-3654118205bd.png)
![ugs03](https://cloud.githubusercontent.com/assets/15223909/11831918/40585290-a381-11e5-89b7-84831f938ce5.png)

**Comment by Denvi** [comment]
Good, one more thing. Response to reset command ("Soft reset" - button in UGS).

**Comment by gregoryperrin** [comment]
it outputs "Grbl 0.9g"

![ugs05](https://cloud.githubusercontent.com/assets/15223909/11832355/71837d32-a385-11e5-9564-7aa456070694.png)

**Comment by Denvi** [comment]
Custom firmware?
```
// Welcome message
void report_init_message()
{
  printPgmString(PSTR("\r\nGrbl " GRBL_VERSION " ['$' for help]\r\n"));
}
```

Should be "Grbl 0.9g ['$' for help]"

**Comment by gregoryperrin** [comment]
No, it´s the Nighlty build, last build available at the moment.

Ok i will try tomorrow, thanks.

**Comment by Denvi** [comment]
I thought that "0.9j" version is the latest. "0.9g" - old one.
https://github.com/grbl/grbl

**Comment by gregoryperrin** [comment]
I thought you were talking about UGS... 

The GRBL version is the one that shipped with my machine. I didn't upgrade it myself.

I'll try your binary tonight.

**Comment by gregoryperrin** [comment]
IT'S WORKING!!!

Nice.

Thanks! What did you change?

**Comment by Denvi** [comment]
Just changed sign, detecting GRBL reset procedure.
Also there were problems with "Abort" function. They were fixed in the latest version (0.8.1).
Take care using "M2" & "M30" commands in your G-code programs. They resets GRBL state, because of what the program doesn't display the "Job done" message.
